packages/core/saved-objects/core-saved-objects-api-server-internal/src/test_helpers/repository.test.common.ts

```diff
@@ -236,54 +253,212 @@ export const enforceError = SavedObjectsErrorHelpers.decorateForbiddenError(
   'User lacks privileges'
 );
 
-export const setupPerformAuthFullyAuthorized = (
-  mockSecurityExt: jest.Mocked<ISavedObjectsSecurityExtension>
+export const setupAuthorizeCreate = (
+  mockSecurityExt: jest.Mocked<ISavedObjectsSecurityExtension>,
+  status: 'fully_authorized' | 'partially_authorized' | 'unauthorized'
+) => {
+  mockSecurityExt.authorizeCreate.mockImplementation(
+    (params: AuthorizeCreateParams): Promise<CheckAuthorizationResult<string>> => {
```

Comment:
Good to know! And in that case, could most of these then be combined into a single method where we pass the jest fn instead of the mock ext?

Reply:
> could most of these then be combined into a single method where we pass the jest fn instead of the mock ext?

If the return value is the same, I guess so, why not.